packages/debugger/src/service.ts

```diff
@@ -327,6 +327,35 @@ export class DebuggerService implements IDebugger, IDisposable {
     }
   }
 
+  /**
+   * Request to set a variable in the global scope.
+   *
+   * @param name The name of the variable.
+   */
+  async copyToGlobals(name: string): Promise<void> {
+    if (!this.session) {
+      throw new Error('No active debugger session');
+    }
+    const frames = this.model.callstack.frames;
+    this.session
+      .sendRequest('copyToGlobals', {
```

Comment:
Since this relies on non-standard method, we need to be very careful here and do one or both:
- detect support (adding `supportsCopyToGlobals`) and suggest this feature upstream and only condition `isEnabled` on support
- enhance the error handling beyond just printing reason to console; we need to tell user that this is requires `copyToGlobals` method which is implemented in IPython >{version number} and xeus-python and to use it they need to upgrade or ask the third-party kernel developer to consider adding support for it.

We also need to document it (in the repository where all other custom methods are listed).

Reply:
Thanks @krassowski.
I already opened a PR to document it: https://github.com/jupyter/jupyter_client/pull/932
It seems difficult to handle the error as the request is always successful (status=200), even if the copy does not occur.
The first suggestion is probably better. I'll see on `ipykernel` and `xeus-python` how to add this information.

Reply:
I opened https://github.com/jupyter/enhancement-proposals/pull/93

Reply:
I added a flag on the `copyToGlobals` support from the kernel.
I also opened https://github.com/ipython/ipykernel/pull/1099 and https://github.com/jupyter-xeus/xeus-python/pull/582.
We will have to wait until the `ipykernel` one is released to be able to make the tests.